lib/engine/game/g_18_al.rb

```diff
@@ -19,6 +19,22 @@ class G18AL < Base
       def operating_round(round_num)
         Round::G18AL::Operating.new(@corporations, game: self, round_num: round_num)
       end
+
+      def revenue_for(route)
+        ensure_terminus_first_or_last(route, %w[A4 Q2])
+
+        super
+      end
+
+      private
+
+      def ensure_terminus_first_or_last(route, termini)
+        termini.each do |terminus|
+          index = route.hexes.index { |hex| hex.name == terminus }
```

Comment:
Cleaner doing
route.hexes[1...-1].map(&:name).include?(terminus)
or even
(route.hexes[1...-1].map(&:name) || termini).empty?
The above removes the need for the loop

(Note the three dots)

Reply:
Now rewritten using & operation for arrays.
Yes, it looks better.